test/src/functions/PolyTestFunction.C

```diff
@@ -34,5 +34,5 @@ PolyTestFunction::PolyTestFunction(const InputParameters & parameters)
 Real
 PolyTestFunction::value(Real /*t*/, const Point & p) const
 {
-  return MathUtils::poly<Real>(_coeffs, p(0), _deriv);
+  return MathUtils::poly(_coeffs, p(0), _deriv);
```

Comment:
Not really testing the new capabilities too much...

Reply:
No, I should probably add a unit test for this, even though the logic doesn't change, and the only think new here would be testing if stuff compiles or not.

Reply:
Ay Chihuahua! The template parameter deduction went horribly wrong here (causing integer return types in some template instantiations). 

Unit test = absofrickinglutely needed

 Fixing now...

Reply:
> Ay Chihuahua!
😄

Reply:
Ok, it wasn't quite that bad, but the new version should be better.